internal/services/storage/storage_share_resource.go

```diff
@@ -143,239 +150,424 @@ func resourceStorageShare() *pluginsdk.Resource {
+		if err != nil {
+			return fmt.Errorf("building File Share Client for %s: %+v", account.StorageAccountId, err)
+		}
+
+		props, err := client.Get(ctx, id.ShareName)
+		if err != nil {
+			return err
+		}
+		if props == nil {
+			log.Printf("[DEBUG] File Share %q was not found in %s - assuming removed & removing from state", id.ShareName, account.StorageAccountId)
+			d.SetId("")
+			return nil
+		}
+
+		d.Set("name", id.ShareName)
+		d.Set("storage_account_name", id.AccountId.AccountName)
+		d.Set("quota", props.QuotaGB)
+		d.Set("url", id.ID())
+		d.Set("enabled_protocol", string(props.EnabledProtocol))
+
+		accessTier := ""
+		if props.AccessTier != nil {
+			accessTier = string(*props.AccessTier)
+		}
+		d.Set("access_tier", accessTier)
+
+		if err := d.Set("acl", flattenStorageShareACLsDeprecated(props.ACLs)); err != nil {
+			return fmt.Errorf("flattening `acl`: %+v", err)
+		}
+
+		if err := d.Set("metadata", FlattenMetaData(props.MetaData)); err != nil {
+			return fmt.Errorf("flattening `metadata`: %+v", err)
+		}
+
+		resourceManagerId := parse.NewStorageShareResourceManagerID(account.StorageAccountId.SubscriptionId, account.StorageAccountId.ResourceGroupName, account.StorageAccountId.StorageAccountName, "default", id.ShareName)
+		d.Set("resource_manager_id", resourceManagerId.ID())
 
-	account, err := storageClient.FindAccount(ctx, subscriptionId, id.AccountId.AccountName)
-	if err != nil {
-		return fmt.Errorf("retrieving Account %q for Share %q: %v", id.AccountId.AccountName, id.ShareName, err)
-	}
-	if account == nil {
-		log.Printf("[WARN] Unable to determine Account %q for Storage Share %q - assuming removed & removing from state", id.AccountId.AccountName, id.ShareName)
-		d.SetId("")
 		return nil
 	}
 
-	// The files API does not support bearer tokens (@manicminer, 2024-02-15)
-	client, err := storageClient.FileSharesDataPlaneClient(ctx, *account, storageClient.DataPlaneOperationSupportingOnlySharedKeyAuth())
+	id, err := fileshares.ParseShareID(d.Id())
 	if err != nil {
-		return fmt.Errorf("building File Share Client for %s: %+v", account.StorageAccountId, err)
+		return err
 	}
 
-	props, err := client.Get(ctx, id.ShareName)
+	existing, err := sharesClient.Get(ctx, *id, fileshares.DefaultGetOperationOptions())
 	if err != nil {
-		return err
+		if response.WasNotFound(existing.HttpResponse) {
+			log.Printf("[DEBUG] %q was not found, removing from state", *id)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %q: %v", *id, err)
 	}
-	if props == nil {
-		log.Printf("[DEBUG] File Share %q was not found in %s - assuming removed & removing from state", id.ShareName, account.StorageAccountId)
-		d.SetId("")
-		return nil
+
+	d.Set("storage_account_id", commonids.NewStorageAccountID(id.SubscriptionId, id.ResourceGroupName, id.StorageAccountName).ID())
+
+	if model := existing.Model; model != nil {
+		d.Set("name", id.ShareName)
```

Comment:
Since we have this information in the `id` and it's not dependent on whether model is `nil` or not we should probably set this outside of this block